components/form/LifecycleHooks.vue

```diff
@@ -0,0 +1,135 @@
+<script>
+import Vue from 'vue';
+
+import HookOption from '@/components/form/HookOption';
+import { _VIEW } from '@/config/query-params';
+
+export default {
+  components: { HookOption },
+
+  props: {
+    mode: {
+      type:     String,
+      required: true,
+    },
+
+    // container spec
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+  },
+
+  data() {
+    return {
+      showHooks:    [],
+      hookOptions:  ['postStart', 'preStop'],
+    };
+  },
+
+  computed: {
+    isView() {
+      return this.mode === _VIEW;
+    },
+  },
+
+  methods: {
+    remove(row) {
+      const idx = this.showHooks.indexOf(row);
+
+      this.showHooks.splice(idx, 1);
+      delete this.value[row];
+      this.update();
+    },
+
+    update() {
+      for (const prop in this.value) {
+        const val = this.value[prop];
+
+        if (val === '' || typeof val === 'undefined' || val === null) {
+          Vue.delete(this.value, prop);
+        } else {
+          Vue.set(this.value, prop, val);
+        }
+      }
+
+      this.$emit('input', this.value);
+    },
+
+    addLifecycleHook(option) {
+      const newHook = { [option]: {} };
+
+      if (!this.checkForHook(option)) {
+        Object.assign(this.value, newHook);
+        this.showHooks.push(option);
+      }
+    },
+
+    checkForHook(opt) {
+      return opt in this.value;
+    }
+  }
+};
+</script>
+
+<template>
+  <div @input="update">
+    <template>
+      <div class="mb-20">
+        <template v-if="showHooks.includes('postStart')">
+          <HookOption :value="value.postStart" :mode="mode" />
+        </template>
+        <div>
+          <template>
+            <button
```

Comment:
Can we just show the two HookOption components all the time, and have them empty (no radio selected) by default? As it is now, when I initially looked at this form with (very) limited knowledge of lifecycle hooks the add/remove buttons made me think I'd be able to add an array of hooks.

Barring that, we need to at least hide the add buttons after they're clicked once. They stay on the page after adding a hook despite no longer doing anything.

Reply:
I changed both HookOptions to always show with no hook selected in the radio buttons by default.